src/Common/src/CommandLine/CommandLineHelpers.cs

```diff
@@ -29,8 +29,11 @@ public static void AppendExpandedPaths(Dictionary<string, string> dictionary, st
                 foreach (string fileName in Directory.EnumerateFiles(directoryName, searchPattern))
                 {
                     string fullFileName = Path.GetFullPath(fileName);
-
-                    string simpleName = Path.GetFileNameWithoutExtension(fileName);
+                    string simpleName = Path.GetFileName(fullFileName);
+                    if (Path.GetExtension(fullFileName) != ".netmodule")
```

Comment:
I need the ".netmodule" extension to get a match later for the "simple name"
`ModuleReference.Name`gives me the full name (including extension .netmodule).
`AssemblyReference` has `AssemblyName` which loses the dll extension.

Reply:
`.netmodule` is just a convention. It's not mandatory.

```
.assembly extern mscorlib {}
.assembly foo {}

.module extern foo.exe

.class MyValue extends [mscorlib]System.ValueType
{
  .field int32 _x
}

.method static void Main()
{
  .entrypoint

  ldtoken valuetype [.module foo.exe]MyValue
  pop
  ret
}
```

(Assemble as `ilasm foo.il`.)

The above works on CLR and passes PEVerify.

Reply:
Sure, I'm aware of that but think it's a valid (but improvable) solution for now. Definitely better than before - i.e. not working at all.

The alternative I considered, would be a instantiating a `PEReader` and `MetadataReader` at that point and checking whether it's an assembly or a .netmodule. I did not really like the redundancy and overhead of this solution, though.

But I am open for ideas. What would be your preferred / suggested way to solve this?

Reply:
Would you approve of this solution?
- Check if specified file is an assembly (e.g. with `System.Reflection.AssemblyName.GetAssemblyName`)
- If yes, remove its extension.
- If not, it is "something else" (maybe a .netmodule) and we add it to the dictionary with name + extension.

Reply:
> it's a valid (but improvable) solution for now

My concern is that things under the `\Common\` directory are shared by many projects/products (some of which don't even live in this repo). If we're adding more support for multi-module assemblies, we need to be sure we nail it because making changes to parts shared by many products is a pain. It's fine if the support is incomplete, we just need to make sure the design that's in `\Common\` what we want to keep. E.g. other projects using `CommandLineHelpers` might not appreciate the differences in behavior based on extension.

This looks more like a problem caused by conflating assembly simple names and file names into a single string dictionary when in fact they're two different things (i.e. it's the simplifications in the ILVerifier project around `ResolveAssemblyOrNetmodule` that are seeping into the `\Common\` directory).